pkg/server/registry/apigroups/acorn/projects/storage.go

```diff
@@ -77,3 +84,40 @@ func (pr *projectCreater) Create(ctx context.Context, object types.Object) (type
 
 	return pr.creater.Create(ctx, object)
 }
+
+func migrateLegacyNamespaces(ctx context.Context, c kclient.Client) error {
+	// Once a project namespace is managed by the project instance, then this label will be set on it.
+	// Therefore, using this label selector ensures the migration only happens once for a namespace.
+	notManaged, err := klabels.NewRequirement(labels.AcornManaged, selection.DoesNotExist, nil)
+	if err != nil {
+		return err
+	}
+
+	nsSelector := klabels.SelectorFromSet(map[string]string{
+		labels.AcornProject: "true",
+	}).Add(*notManaged)
+	namespaces := corev1.NamespaceList{}
+	if err := c.List(ctx, &namespaces, kclient.MatchingLabelsSelector{Selector: nsSelector}); err != nil {
+		return err
+	}
+
+	for _, ns := range namespaces.Items {
+		project := &v1.ProjectInstance{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ns.Name,
+			},
+			Spec: v1.ProjectInstanceSpec{
+				DefaultRegion: ns.Annotations[labels.AcornProjectDefaultRegion],
+			},
+		}
+
+		if supportedRegions := ns.Annotations[labels.AcornProjectSupportedRegions]; supportedRegions != "" {
+			project.Spec.SupportedRegions = strings.Split(supportedRegions, ",")
+		}
+		if err = c.Create(ctx, project); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
If the project store isn't up and running until after `migrateLegacyNamespaces` returns, how can this client talk to the Project API? Does it hit the "old" API of the pod upgraded from? If that's the case, it doesn't seem like this would work as a net-new deployment.

Reply:
This create call is hitting using a `ProjectInstance`, which is stored in Kubernetes.

Reply:
Confirmed I am able to run `acorn install` with these changes on a fresh Kubernetes cluster.